configuration/marathon.go

```diff
@@ -9,13 +11,56 @@ import (
 */
 type Marathon struct {
 	// comma separated marathon http endpoints including port number
-	Endpoint string
-	User     string
-	Password string
-
+	Endpoint       string
+	UseZookeeper   bool
+	Zookeeper      Zookeeper
+	User           string
+	Password       string
 	UseEventStream bool
 }
 
 func (m Marathon) Endpoints() []string {
+	if m.UseZookeeper {
+		endpoints, err := zkEndpoints(m.Zookeeper)
+		if err != nil {
+			return nil
```

Comment:
If we can't return an error, let's log it. This is a fairly important error case.